app/models/events/publishing_api.rb

```diff
@@ -1,4 +1,7 @@
 class Events::PublishingApi < ApplicationRecord
-  self.table_name = "publishing_api_events"
+  self.table_name = "publishing_api_events" # this line seems to be problematic
+  # We get the following error:
+  #    #<ActiveModel::MissingAttributeError: can't write unknown attribute `publishing_api_id`>
+  # ...which suggests Rails / ActiveRecord is looking for the default "#{class name}_id" key, which we've overridden here.
```

Comment:
Possibly related: https://github.com/rails/rails/issues/28192
Also possibly related: https://github.com/rails/rails/issues/42656
Also read this: https://www.bigbinary.com/blog/rails-6-1-adds-support-for-belongs_to-to-has_many-inversing